test/runtests.jl

```diff
@@ -369,6 +369,23 @@ end
             test_inferred(bm, TVI, (v3,  3))
             test_inferred(bp, TVF, (v3, .3))
             test_inferred(bm, TVF, (v3, .3))
+            @testset "views" begin
+                va = @view TVI(1:3)[1:2]
+                vb = @view TVI(4:6)[2:3]
+                vf2 = TVF(7:8)
+                test_inferred(+,  TVI, (va,  vb))
+                test_inferred(+,  TVF, (va, vf2))
+                test_inferred(+,  TVF, (vf2, vb))
```

Comment:
Interestingly, the combination of a view and a materialised FixedSizeArray already returned FixedSizeArray before this PR.  I feel like Base should automatically handle the SubArray case the same way the underlying array type is broadcast, instead of requesting downstream packages to do it.

Reply:
Yeah. Both `Array` and `SubArray` just dispatch to the fallback method:

```julia-repl
julia> @which Base.BroadcastStyle(Vector{Float32})
Base.Broadcast.BroadcastStyle(::Type{<:AbstractArray{T, N}}) where {T, N}
     @ Base.Broadcast broadcast.jl:102

julia> typeof(@view [10, 20, 30, 40][2:3])
SubArray{Int64, 1, Vector{Int64}, Tuple{UnitRange{Int64}}, true}

julia> @which Base.BroadcastStyle(ans)
Base.Broadcast.BroadcastStyle(::Type{<:AbstractArray{T, N}}) where {T, N}
     @ Base.Broadcast broadcast.jl:102
```

Presumably an additional method for `SubArray` would be good.